project_euler/problem_082/sol1.py

```diff
@@ -0,0 +1,65 @@
+"""
+Project Euler Problem 82: https://projecteuler.net/problem=82
+
+The minimal path sum in the 5 by 5 matrix below, by starting in any cell
+in the left column and finishing in any cell in the right column,
+and only moving up, down, and right, is indicated in red and bold;
+the sum is equal to 994.
+
+     131    673   [234]  [103]  [18]
+    [201]  [96]   [342]   965    150
+     630    803    746    422    111
+     537    699    497    121    956
+     805    732    524    37     331
+
+Find the minimal path sum from the left column to the right column in matrix.txt
+(https://projecteuler.net/project/resources/p082_matrix.txt)
+(right click and "Save Link/Target As..."),
+a 31K text file containing an 80 by 80 matrix.
+"""
+
+import os
+
+
+def solution(filename: str = "p082_matrix.txt") -> int:
+    """
+    Returns the minimal path sum in the matrix from the file, by starting in any cell
+    in the left column and finishing in any cell in the right column,
+    and only moving up, down, and right
+
+    >>> solution("test_matrix.txt")
+    994
+    """
+
+    with open(os.path.join(os.path.dirname(__file__), filename)) as input_file:
+        matrix = [
+            [int(element) for element in line.split(",")]
+            for line in input_file.readlines()
+        ]
+
+    m = len(matrix)
+    n = len(matrix[0])
```

Comment:
```suggestion
    rows = len(matrix)
    cols = len(matrix[0])
```
I think this would be better for posterity. Please update wherever else is required.

Reply:
Renamed